libs/sdk/src/decorators/tool.decorator.ts

```
@@ -48,4 +48,92 @@ export {
+      __Err<"execute return type must be output schema or Promise<output schema>">;
+
+  // Rewrapped constructor: preserve concrete/abstract + original type, but guarantee ToolContext<In,Out> on instance
+  type __Rewrap<C extends __Ctor, In, Out> =
+    C extends abstract new (...a: __A<C>) => __R<C>
+      ? C & (abstract new (...a: __A<C>) => (ToolContext<In, Out> & __R<C>))
+      : C extends new (...a: __A<C>) => __R<C>
+        ? C & (new (...a: __A<C>) => (ToolContext<In, Out> & __R<C>))
+        : never;
+
+  // ---------- the decorator ----------
+  // @ts-ignore
```

ESLint

[error] Use "@ts-expect-error" instead of "@ts-ignore", as "@ts-ignore" will do nothing if the following line is error-free.

(@typescript-eslint/ban-ts-comment)